test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -245,9 +307,28 @@ private TestDirectory CopyLatestCliForPack()
         private void CopyLatestCliToTestDirectory(string destinationDir)
         {
             var cliDir = Path.GetDirectoryName(_dotnetCli);
+            var sdkDir = Path.Combine(cliDir, "sdk" + Path.DirectorySeparatorChar);
+
+            // Determine which SDK version to copy
+            SdkVersion = GetSdkToTest(sdkDir);
+
+            WriteGlobalJson(destinationDir);
 
-            //Create sub-directory structure in destination
-            foreach (var directory in Directory.GetDirectories(cliDir, "*", SearchOption.AllDirectories))
+            var sdkPath = Path.Combine(sdkDir, SdkVersion + Path.DirectorySeparatorChar);
+            var fallbackFolderPath = Path.Combine(sdkDir, "NuGetFallbackFolder");
+
+            Func<string, bool> predicate = path =>
+            {
+                if (!path.StartsWith(sdkDir))
+                {
+                    return true;
+                }
+
+                return path.StartsWith(sdkPath) || path.StartsWith(fallbackFolderPath);
+            };
+
+            //Create sub-directory structure in destination, ignoring any SDK version not selected.
+            foreach (var directory in Directory.GetDirectories(cliDir, "*", SearchOption.AllDirectories).Where(predicate))
```

Comment:
As per doc [`Directory.EnumerateDirectories`](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.enumeratedirectories?view=net-5.0), is efficient over `Directory.GetDirectories` because we can start enumerating the collection of names before the whole collection is returned. I remember Damon gave this feedback in one of my PRs to use `EnumerateDirectories`. I see couple of occurrences of `Directory.GetDirectories` in this PR.